### PR TITLE
IgxTree Samples

### LINKED ISSUE #2603: Samples for igxTree Component
### Description  
Create two samples for the new igxTree component based on [this](https://app.abstract.com/projects/508d16c1-5e6e-4f03-9956-e3cf27c87fa5/branches/8bb1943c-c076-41c1-a4ec-f44f6ba7a179/collections/924392b8-4765-4ec6-a7db-dae62312abaa) design 

Sample 1: Basic sample illustrating how to initialize igxTree.
Sample 2: Sample illustrating tree`s functionalities (selection, node templating, load on demand).

### PR DESCRIPTION
**Pull request by MonikaKirkova**
closes #2603

### CODE REVIEW

**Review by kacheshmarova — DISMISSED** [review]

File: src/app/lists/tree/tree-advanced-sample/services/data.service.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (kdinev) [inline src/app/lists/tree/tree-advanced-sample/services/data.service.ts:1]: Please move the service under `./services`.

File: src/app/lists/tree/tree-advanced-sample/local-data.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (kdinev) [inline src/app/lists/tree/tree-advanced-sample/local-data.ts:1]: Please move the data under `./data`.

File: src/app/lists/tree/tree-basic-sample/local-data.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (kdinev) [inline src/app/lists/tree/tree-basic-sample/local-data.ts:1]: Please move the data under `./data`.

**Review by kdinev — CHANGES_REQUESTED** [review]

File: src/app/lists/tree/tree-advanced-sample/services/data.service.ts (hunk 1 above)
Comment (ViktorSlavov) [reply src/app/lists/tree/tree-advanced-sample/services/data.service.ts:1]: @kdinev 
To be clear, these should be moved under `app/{featureModule}/services` folder (in this case, `app/lists/services`) as opposed to `app/services` (which is a separate feature module)?

Comment (kdinev) [reply src/app/lists/tree/tree-advanced-sample/services/data.service.ts:1]: You can leave the service as is, but the data should be moved to the data folder.

Comment (ViktorSlavov) [reply src/app/lists/tree/tree-advanced-sample/services/data.service.ts:1]: Thanks for clarifying! Data's now moved to the root (`app/data`) folder

**Review by kdinev — APPROVED** [review]

**Review by ViktorSlavov — APPROVED** [review]

**Pull request merged**